referenceframe/frame.go

```diff
@@ -166,6 +166,24 @@ type staticFrame struct {
 	geometryCreator spatial.GeometryCreator
 }
 
+// a tailGeometryStaticFrame is a static frame whose geometry is placed at the end of the frame's transform, rather than at the beginning.
+type tailGeometryStaticFrame struct {
```

Comment:
[q] why is this being introduced?

Reply:
If a component has a geometry, we put that in the "_origin" frame rather than the actual component frame, because for things like arms, the actual component frame is the arm model itself.

The other option would be to make a new type of frame that wraps frames, but then we'd need to marshal that